chain/index/README.MD

```diff
@@ -0,0 +1,188 @@
+[ChainIndexer]
+  GCRetentionEpochs = X  # Replace X with your desired value
+```
+
+- Setting `GCRetentionEpochs` to 0 (**this is the default**) completely disables GC.
+- Any non-zero value enables GC and determines the number of epochs of historical data to retain.
+
+#### Recommendations
+
+1. **Archival Nodes**: **Keep GC disabled** (`GCRetentionEpochs` = 0) to retain all indexed data.
+
+2. **Non-Archival Nodes**:  Set `GCRetentionEpochs` to match the amount of chain state your node retains 
+(*for example:* if your node is configured to retain 2 days of Filecoin chain state with the Splitstore, set `GCRetentionEpochs` to (number of Filecoin epochs in a day *2) = 5760).
+
+---
+
+**Note: The following config options no longer exist in Lotus and have been removed in favor of the `ChainIndexer` config options explained above:**
+
+```toml
+[Fevm]
+EthTxHashMappingLifetimeDays = 0
+
+[Events]
+DisableHistoricFilterAPI = false
+DatabasePath = ""
+```
+
+These options are now deprecated and will not have any effect if used in your configuration file. Please use the `ChainIndexer` config options as described above.
+
+---
+
+## Migration Guide
+
+Migrating to the new `ChainIndexer` involves several steps to ensure a smooth transition. Here's a guide to help you through the process:
+
+1. **Backup Existing Index Databases**
+   - Before restarting your Lotus node, create a backup of your existing index databases.
+   - These files are located in the `{$LOTUS_PATH/sqlite}` directory.
+   - While not used in the migration process, these backups are crucial for potential rollbacks.
+
+2. **Remove Old Index Files**
+   - After creating backups, remove the SQLite database files for `MsgIndex`, `EthTxIndex`, and `EventIndex` from the `{$LOTUS_PATH/sqlite}` directory.
+
+3. **Update Configuration**
+   - Modify your Lotus configuration to enable the `ChainIndexer` as described in the `ChainIndexer Config` section above.
+
+4. **Restart Lotus Node**
+   - Restart your Lotus node with the new configuration.
+   - The `ChainIndexer` will begin indexing **real-time chain state changes** immediately.
+
+Once Lotus starts with the `ChainIndexer` enabled, it will begin indexing real-time chain state changes i.e. new incoming tipsets. However, it will not index any historical chain state i.e. any previously existing chain state. To index historical chain state (aka **"backfilling"**), you can use the following tools that we're shipping with the `ChainIndexer`:
+
+### The `ChainValidateIndex` JSON RPC API
+
+The `ChainValidateIndex` JSON RPC API serves a dual purpose: it validates/diagnoses the integrity of the index at a specific epoch (i.e., it ensures consistency between indexed data and actual chain state), while also providing the option to backfill the `ChainIndexer` if it does have data for the specified epoch. 
+
+```go
+// IndexValidation contains detailed information about the validation status of a specific chain epoch.
+type IndexValidation struct {
```

Comment:
I assume this is all copy/paste from source code.  I worry it will get out of date.  Can we link to the canonical source that has this information?  Should we link to api-v1-unstable-methods.md and make sure all these comments show up there?

Also, we're talking about this in terms of Go code, but consumers of this are thinking about this in terms of json-rpc right?  Can we talk about it in terms of json-rpc types?